// @fluentui/react-component-ref/dist/dts/src/RefFindNode.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var React = require("react");
var ReactDOM = require("react-dom");
var utils_1 = require("./utils");
var RefFindNode = /** @class */ (function (_super) {
    tslib_1.__extends(RefFindNode, _super);
    function RefFindNode() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.prevNode = null;
        return _this;
    }
    RefFindNode.prototype.componentDidMount = function () {
        this.prevNode = ReactDOM.findDOMNode(this);
        utils_1.handleRef(this.props.innerRef, this.prevNode);
    };
    RefFindNode.prototype.componentDidUpdate = function (prevProps) {
        var currentNode = ReactDOM.findDOMNode(this);
        if (this.prevNode !== currentNode) {
            this.prevNode = currentNode;
            utils_1.handleRef(this.props.innerRef, currentNode);
        }
        if (prevProps.innerRef !== this.props.innerRef) {
            utils_1.handleRef(this.props.innerRef, currentNode);
        }
    };
    RefFindNode.prototype.componentWillUnmount = function () {
        utils_1.handleRef(this.props.innerRef, null);
        delete this.prevNode;
    };
    RefFindNode.prototype.render = function () {
        var children = this.props.children;
        return children;
    };
    return RefFindNode;
}(React.Component));
exports.RefFindNode = RefFindNode;
